### DISCUSSION TITLE
Checklist on RichTextEditor (RTE)

### DISCUSSION THREAD

**Discussion by suchintan** [mantinedev/mantine#1722] [Feature requests]
Quill seems to have support for checklists, either via a module (https://github.com/koffeinfrei/quill-task-list) or officially (https://github.com/quilljs/quill/issues/759)

Is it possible to add this feature into the RichTextEditor? Would be very useful!

If not, I can't quite figure out how to add custom modules into RichTextEditor, particularly to get `quill-task-list` working. Would love some support if possible

**Comment by rtivital** [comment]
`@mantine/rte` does not support a lot of customizations, it is designed to be as simple as possible. If you need custom features that are not included, you should build your own editor with react-quill or similar library.

**Comment by suchintan** [comment]
That makes sense @rtivital. I'd like to still request the addition of a checkbox into the RTE. It seems like a pretty useful feature that many rich text editors support, even simpler tools like Markdown.

**Comment by suchintan** [comment]
@rtivital do you know if you'll include this in the eventual 6.0 release?

**Reply by rtivital** [reply]
Most likely in 6.0 rte package will be migrated to tiptap, you will be able to use any plugins that tip tap provides

**Reply by suchintan** [reply]
That's awesome news! Tiptap seems like a cool library. Do you know when 6.0 is gonna be released?

**Reply by rtivital** [reply]
Probably in May

**Comment by Brian-McBride** [comment]
@suchintan I've been working on a text editor last week. I'll say that the codebase that rtivital has for the RTE is very easy to follow, and it wouldn't be very difficult to add checkboxes. 

I am not ready to release to NPM, but I am working on an RTE using Slate and Mantine. Hopefully, I can share out a beta version later this week. I need some checkbox support as well.

**Comment by suchintan** [comment]
That's great @Brian-McBride

Tip tap or remirror seems to be another promising option for a WYSIWYG editor. I was thinking about building my own integration soon
